core/lib/rom/schema.rb

```diff
@@ -42,6 +42,7 @@ class Schema
       registry.each do |_, relation|
         unless relation.schema.frozen?
           relation.schema.finalize_associations!(relations: registry)
+          relation.schema.finalize_relations!(relations: registry)
```

Comment:
Instead of doing this, it'd be better to [tweak this line](https://github.com/rom-rb/rom/blob/master/core/lib/rom/setup/finalize/finalize_relations.rb#L106) and do:

``` ruby
options = { __registry__: registry, mappers: mappers, schema: schema.with(relations: registry), **plugin_options }
```